samples/snippets/csharp/VS_Snippets_Misc/tpl_parallel/cs/simpleforeach.cs

```diff
@@ -1,40 +1,112 @@
 ﻿// <snippet03>
 using System;
-using System.IO;
-using System.Threading;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
 using System.Threading.Tasks;
-using System.Drawing;
 
-public class Example
+namespace ParallelExample
 {
-    public static void Main()
+    class Program
     {
-        // A simple source for demonstration purposes. Modify this path as necessary.
-        string[] files = Directory.GetFiles(@"C:\Users\Public\Pictures\Sample Pictures", "*.jpg");
-        string newDir = @"C:\Users\Public\Pictures\Sample Pictures\Modified";
-        Directory.CreateDirectory(newDir);
-
-        // Method signature: Parallel.ForEach(IEnumerable<TSource> source, Action<TSource> body)
-        Parallel.ForEach(files, (currentFile) =>
-                                {
-                                    // The more computational work you do here, the greater
-                                    // the speedup compared to a sequential foreach loop.
-                                    string filename = Path.GetFileName(currentFile);
-                                    var bitmap = new Bitmap(currentFile);
-
-                                    bitmap.RotateFlip(RotateFlipType.Rotate180FlipNone);
-                                    bitmap.Save(Path.Combine(newDir, filename));
-
-                                    // Peek behind the scenes to see how work is parallelized.
-                                    // But be aware: Thread contention for the Console slows down parallel loops!!!
-
-                                    Console.WriteLine($"Processing {filename} on thread {Thread.CurrentThread.ManagedThreadId}");
-                                    //close lambda expression and method invocation
-                                });
-
-        // Keep the console window open in debug mode.
-        Console.WriteLine("Processing complete. Press any key to exit.");
-        Console.ReadKey();
+        static void Main(string[] args)
+        {
+            // 2 million
+            int limit = 2 * 1000000;
+            var inputs = new List<int>(limit);
+            Random radomGenerator = new Random();
+            for (int index = 0; index < limit; index++)
+            {
+                inputs.Add(radomGenerator.Next());
+            }
+
+            var watch = new Stopwatch();
+            watch.Start();
+            var primeNumbers = GetPrimeList(inputs);
+            watch.Stop();
+
+            var watchForParallel = new Stopwatch();
+            watchForParallel.Start();
+            var primeNumbersFromParallel = GetPrimeListWithParallel(inputs);
+            watchForParallel.Stop();
```

Comment:
```suggestion
            var watchForParallel = Stopwatch.StartNew();
            var primeNumbersFromParallel = GetPrimeListWithParallel(numbers);
            watchForParallel.Stop();
```

Use `StartNew` as it instantiates a `Stopwatch`, starts, and returns it.

Reply:
Make sense. Updated.